ethers-solc/src/cache.rs

```diff
@@ -153,10 +151,11 @@ impl SolFilesCache {
     pub fn has_changed_or_missing_artifact<T: ArtifactOutput>(
         &self,
         file: &Path,
-        hash: &[u8],
+        hashes: &HashMap<PathBuf, String>,
         config: Option<&SolcConfig>,
-        artifacts_root: &Path,
+        paths: &ProjectPathsConfig,
     ) -> bool {
+        let hash = hashes.get(file).unwrap().as_bytes();
```

Comment:
This `unwrap` should be safe if used with `get_changed_or_missing_artifacts_files`, but could panic if somebody did not construct the map properly, should we handle the `None` case, or should we restrict the visibility of this function so it can never be called in an unsafe way?

Reply:
good catch, yes there's not guarantee, will revise